napari/_qt/qt_about_keybindings.py

```diff
@@ -1,105 +1,80 @@
-        self.active_label.setSizePolicy(
-            QSizePolicy.Expanding, QSizePolicy.Expanding
-        )
-
-        self.update_text(None)
+        self.layout = QVBoxLayout()
+        self.setLayout(self.layout)
 
-        scroll_widget = QWidget()
-        scroll_widget.setSizePolicy(
-            QSizePolicy.Expanding, QSizePolicy.Expanding
+        # stacked keybindings widgets
+        self.textEditBox = QTextEdit()
+        self.textEditBox.setTextInteractionFlags(Qt.TextSelectableByMouse)
+        self.textEditBox.setMinimumWidth(360)
+        self.keybindings_strs = {'All active keybindings': ''}
+        layers = [
+            napari.layers.Image,
+            napari.layers.Labels,
+            napari.layers.Points,
+            napari.layers.Shapes,
+            napari.layers.Surface,
+            napari.layers.Vectors,
+        ]
+        for layer in layers:
+            if len(layer.class_keymap) == 0:
+                text = 'No keybindings'
+            else:
+                text = get_keybindings_summary(layer.class_keymap)
+            self.keybindings_strs[f"{layer.__name__} layer"] = text
+
+        # layer type selection
+        self.layerTypeComboBox = QComboBox()
+        for name in self.keybindings_strs.keys():
+            self.layerTypeComboBox.addItem(name)
+        self.layerTypeComboBox.activated[str].connect(
+            lambda text=self.layerTypeComboBox: self.change_layer_type(text)
         )
-        scroll_widget.setContentsMargins(10, 10, 10, 10)
-        scroll_layout = QVBoxLayout()
-        scroll_layout.addWidget(self.active_label)
-        scroll_layout.addStretch(1)
-        scroll_widget.setLayout(scroll_layout)
-        self.setWidget(scroll_widget)
-
-        self.viewer.events.active_layer.connect(self.update_text)
-
-    def update_text(self, event):
-        keybindings_str = ''
+        current_layer = 'All active keybindings'
+        self.layerTypeComboBox.setCurrentText(current_layer)
+        # self.change_layer_type(current_layer)
+        layer_type_layout = QHBoxLayout()
+        layer_type_layout.setContentsMargins(10, 5, 0, 0)
+        layer_type_layout.addWidget(self.layerTypeComboBox)
+        layer_type_layout.addStretch(1)
+        layer_type_layout.setSpacing(0)
+        self.layout.addLayout(layer_type_layout)
+        self.layout.addWidget(self.textEditBox, 1)
+
+        self.viewer.events.active_layer.connect(self.update_active_layer)
+        self.update_active_layer(None)
+
+    def change_layer_type(self, text):
+        self.textEditBox.setHtml(self.keybindings_strs[text])
+
+    def update_active_layer(self, event):
+        text = ''
         # Add class and instance viewer keybindings
-        keybindings_str += get_keybindings_summary(self.viewer.class_keymap)
-        keybindings_str += get_keybindings_summary(self.viewer.keymap)
+        text += get_keybindings_summary(self.viewer.class_keymap)
+        text += get_keybindings_summary(self.viewer.keymap)
```

Comment:
what happens if a class keymap and a layer keymap share the same key combination?

Reply:
hmm, right now both will appear (which is same as before this PR). What happens in the code again? We should make sure that the same thing prints as happens, but maybe we do that in a followup PR